ddl/column_type_change_test.go

```diff
@@ -461,6 +469,38 @@ func (s *testColumnTypeChangeSuite) TestColumnTypeChangeBetweenVarcharAndNonVarc
 	tk.MustGetErrCode("alter table t change column a a varchar(10);", mysql.ErrUnsupportedDDLOperation)
 	tk.MustGetErrCode("alter table t change column b b char(10);", mysql.ErrUnsupportedDDLOperation)
 	tk.MustExec("admin check table t;")
+
+	tk.Se.GetSessionVars().EnableChangeColumnType = true
+	tk.MustExec("drop table if exists t;")
+	tk.MustExec("create table t(a char(10), b varchar(10));")
+	tk.MustExec("insert into t values ('aaa    ', 'bbb   ');")
+	tk.MustExec("alter table t change column a a varchar(10);")
+	tk.MustExec("alter table t change column b b char(10);")
+
+	tk.MustExec("alter table t add index idx_a(a);")
+	tk.MustExec("alter table t add index idx_b(b);")
+	tk.MustExec("alter table t change column a a char(10);")
+	tk.MustExec("alter table t change column b b varchar(10);")
+	tk.MustExec("admin check table t;")
```

Comment:
I checkout 8fca6a21a675908cb82c5ac9a69d677fb0fd974a(before PR #23688), and this test will pass. So do we need to delete this test?